controllers/multiclusterlease_controller.go

```diff
@@ -112,14 +112,16 @@ func (r *MultiClusterLeaseReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	leaseInfo, err := leaderElector.AcquireOrRenew(ctx, &mcl, identity)
 	if err != nil {
 		log.Error(err, "failed to acquire or renew lease",
-			"candidate", identity,
-			"currentHolder", leaseInfo.HolderIdentity)
+			"candidate", identity)
 
 		// Update condition to indicate backend is unhealthy
 		r.setBackendHealthyCondition(&mcl, false, err)
 
-		// Update status
-		r.setMCLStatus(&mcl, leaseInfo)
+		// Update status if leaseInfo is not nil
+		if leaseInfo != nil {
+			r.setMCLStatus(&mcl, leaseInfo)
+		}
```

Comment:
Nit: we probably want to be careful what the leaseInfo looks like in the error case, rather than just leaving it as it was, but ... not a blocker